substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -1395,6 +1395,84 @@ where
 		}
 	}
 
+	/// Fetches the 'provides' tags associated to transactions part of blocks
+	/// of an enacted fork.
+	///
+	/// The 'provides' tags of transactions from enacted blocks are searched
+	/// in inactive views. These tags serve as cache based on the inactive views,
+	/// since prunning transactions is based on their 'provides' tags, which if
+	/// not found in the enacted fork tip active view, then they will be obtained
+	/// by revalidating the transaction, which takes time.
```

Comment:
I simplified and improved a flow a bit:
```suggestion
	/// The 'provides' tags of transactions from enacted blocks are searched
	/// in inactive views. Found `provide` tags are intended to serve as cache,
	/// helping to avoid unnecessary revalidations during pruning.
```